src/flow/ui/FlowStackNavigator.tsx

```diff
@@ -1,11 +1,36 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 
-import { FLOW_STACK_SCREEN_ROUTES, FLOW_STACK_SCREENS, INITIAL_FLOW_STACK_SCREEN_ROUTE } from '../nav/FlowStackRoutes';
+import InitialRouteNameState from '../../navigation/InitialRouteNameState';
+
+import { FLOW_STACK_SCREEN_ROUTES, FLOW_STACK_SCREENS, FlowStackRoute } from '../nav/FlowStackRoutes';
+import useFlowStateContext from '../sm/useFlowStateContext';
 import FlowStack from './FlowStack';
 
-const FlowStackNavigator = () => {
+export const FlowStackNavigator = () => {
+  const [initialRouteName, setInitialRouteName] = useFlowStateContext('initialRouteName');
+
+  useEffect(() => {
+    InitialRouteNameState.onChange<FlowStackRoute>((payload) => {
+      setInitialRouteName(payload.value);
+    });
+
+    return () => {
+      InitialRouteNameState.offChange();
+    };
+    // Once during mount only.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  // When the initial route name is still null, don't return anything.
+  if (!initialRouteName) {
+    return null;
+  }
+
+  // Every time the initial route name changes to a different string, reconstruct the entire thing.
+  // Why? The initial route name will only be set once(!), i.e. during the initial render.
+  // Subsequent changes won't alter the initial route.
   return (
-    <FlowStack.Navigator initialRouteName={INITIAL_FLOW_STACK_SCREEN_ROUTE}>
+    <FlowStack.Navigator initialRouteName={initialRouteName}>
       {FLOW_STACK_SCREEN_ROUTES.map((route) => {
         const screen = FLOW_STACK_SCREENS[route];
         return (
```

Comment:
# Remount Stack Navigator technique

The Stack Navigator only sets the initial route once in its entire lifetime, at its initial render, then never again. When `initialRouteName` is `null`, unmount the Stack Navigator so that, on a valid `string` value, the next Stack Navigator `initialRouteName` can be different on the next mount.

## References

This post: https://stackoverflow.com/a/68426073/11455106